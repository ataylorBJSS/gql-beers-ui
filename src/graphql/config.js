import ApolloClient from 'apollo-boost'

const client = new ApolloClient({
    uri: 'http://localhost:3000/graphql',
    request: async operation => {
        const token = 'TEST TOKEN'
        operation.setContext({
            headers: {
                authorization: token,
            },
        })
    },
})

export default client
