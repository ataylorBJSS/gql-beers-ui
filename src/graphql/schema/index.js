import gql from 'graphql-tag'

export const QRY_BEERS_LIST = gql`
    {
        beers {
            id
            name
            tagline
            image_url
            ratings {
                total {
                    average
                    count
                }
            }
        }
    }
`

export const QRY_BEER = gql`
    query Beer($id: Int!) {
        beer(id: $id) {
            id
            name
            tagline
            abv
            image_url
            description
            first_brewed
            food_pairing
            ratings {
                total {
                    average
                    count
                }
                rating {
                    rating
                    comment
                }
            }
        }
    }
`

export const ADD_RATING = gql`
    mutation AddRating($parameters: RatingParameters) {
        addRating(parameters: $parameters) {
            average
            count
        }
    }
`
