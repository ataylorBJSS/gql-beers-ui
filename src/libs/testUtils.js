import { wait } from 'react-testing-library'

export * from 'react-testing-library'

const waitUntilLoadingIsFinished = queryByText =>
    wait(() => {
        const isLoading = queryByText('Loading') != null
        expect(isLoading).toBe(false)
    })

export { waitUntilLoadingIsFinished }
