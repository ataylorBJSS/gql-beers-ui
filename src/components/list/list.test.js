import React from 'react'
import { render, cleanup, waitUntilLoadingIsFinished } from '../../libs/testUtils'
import { MockedProvider } from 'react-apollo/test-utils'
import { BeerList } from './index'
import { QRY_BEERS_LIST } from '../../graphql/schema'

const mocks = [
    {
        request: {
            query: QRY_BEERS_LIST,
        },
        result: {
            data: {
                beers: [
                    {
                        id: '1',
                        name: 'name',
                        abv: 4.2,
                        ebc: 10,
                        srm: 5.4,
                        tagline: 'tagline',
                        image_url: '_IMAGEURL_',
                        ratings: {
                            total: {
                                average: 4,
                                count: 3,
                            },
                        },
                    },
                ],
            },
        },
    },
]

afterEach(cleanup)

it('renders without crashing', () => {
    render(
        <MockedProvider mocks={mocks} addTypename={false}>
            <BeerList />
        </MockedProvider>
    )
})

it('renders loading', () => {
    const { container } = render(
        <MockedProvider mocks={mocks} addTypename={false}>
            <BeerList />
        </MockedProvider>
    )

    expect(container).toMatchSnapshot()
})

it('renders data', async () => {
    const { queryByText, container } = render(
        <MockedProvider mocks={mocks} addTypename={false}>
            <BeerList />
        </MockedProvider>
    )

    await waitUntilLoadingIsFinished(queryByText)

    expect(container).toMatchSnapshot()
})
