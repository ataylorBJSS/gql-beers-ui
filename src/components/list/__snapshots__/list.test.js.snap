// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders data 1`] = `
<div>
  <div
    class="listContainer"
  >
    <div
      class="listItem"
    >
      <div>
        <img
          alt="name"
          src="_IMAGEURL_"
        />
      </div>
      <div>
        name
      </div>
      <div>
        tagline
      </div>
      <div
        class="abv beer-color-10"
      >
        4.2
        %
      </div>
      <div
        class="ratingsWrapper"
      >
        <span
          style="display: inline-block; direction: ltr;"
        >
          <span
            style="cursor: pointer; display: inline-block; position: relative;"
          >
            <span
              style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
            />
          </span>
          <span
            style="cursor: pointer; display: inline-block; position: relative;"
          >
            <span
              style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
            />
          </span>
          <span
            style="cursor: pointer; display: inline-block; position: relative;"
          >
            <span
              style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
            />
          </span>
          <span
            style="cursor: pointer; display: inline-block; position: relative;"
          >
            <span
              style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
            />
          </span>
          <span
            style="cursor: pointer; display: inline-block; position: relative;"
          >
            <span
              style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0%;"
            />
          </span>
        </span>
        <div
          class="counter"
        >
          (
          3
          )
        </div>
      </div>
      <button>
        View Details
      </button>
    </div>
  </div>
</div>
`;

exports[`renders loading 1`] = `
<div>
  <p>
    Getting some beers.....
  </p>
</div>
`;
